### ISSUE TITLE
Error when Python is installed to a path with a space

### ISSUE THREAD

**Issue by koverholt** [rstudio/rsconnect-jupyter#250]
OS: Windows Server 2019
Python Version: 3.7.7
rsconnect-jupyter version: 1.3.1.2

## Description

The `rsconnect-jupyter` extension fails to generate a manifest when Python is installed to a path that includes a space.

## Steps to reproduce

1. Install Python from python.org to a path with spaces (e.g., `C:\Users\Administrator\Desktop\python install`)

2. Install pip
    ```
    curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py
    C:\Users\Administrator\Desktop\python install\python.exe get-pip.py
    ```

3. Install jupyter and extensions, and enable extensions
    ```
    C:\Users\Administrator\Desktop\python install\Scripts\pip install jupyter rsconnect-jupyter
    C:\Users\Administrator\Desktop\python install\Scripts\jupyter-nbextension install --sys-prefix --py rsconnect_jupyter
    C:\Users\Administrator\Desktop\python install\Scripts\jupyter-nbextension enable --sys-prefix --py rsconnect_jupyter
    C:\Users\Administrator\Desktop\python install\Scripts\jupyter-serverextension enable --sys-prefix --py rsconnect_jupyter
    ```

4. Start Jupyter Notebook
    ```
    C:\Users\Administrator\Desktop\python install\Scripts\jupyter.exe notebook
    ```

5. Open Javascript console

6. Click on notebook extension, then "Create Manifest". Then click "Create Manifest" button.

7. The following errors will appear in the Javascript console:

```
menubar.js:263 actions jupyter-notebook:find-and-replace does not exist, still binding it in case it will be defined later...
MenuBar.bind_events @ menubar.js:263
utils.js:60 load_extensions Arguments(2)
bidi.js:19 Loaded moment locale en
session.js:54 Session: kernel_created (b4cd41e4-5e91-4e28-a700-b06a50c6aa29)
kernel.js:462 Starting WebSockets: ws://localhost:8888/api/kernels/ea115100-eedf-43aa-8f94-90b8b83724e2
utils.js:37 Loading extension: jupyter-js-widgets/extension
utils.js:37 Loading extension: rsconnect_jupyter/index
kernel.js:106 Kernel: kernel_connected (ea115100-eedf-43aa-8f94-90b8b83724e2)
kernel.js:106 Kernel: kernel_ready (ea115100-eedf-43aa-8f94-90b8b83724e2)
rsconnect.js?v=20200327211635:252 Using python: c:\users\administrator\desktop\python install\python.exe
rsconnect.js?v=20200327211635:276 executing: !c:\users\administrator\desktop\python install\python.exe -m rsconnect.environment ${PWD}
rsconnect.js?v=20200327211635:12 RSConnect API: environment: 'c:\users\administrator\desktop\python' is not recognized as an internal or external command,
operable program or batch file.

rsconnect.js?v=20200327211635:12 RSConnect API: environment error: SyntaxError: Unexpected token ' in JSON at position 0
    at JSON.parse (<anonymous>)
    at handle_output (rsconnect.js?v=20200327211635:294)
    at Kernel._handle_output_message (kernel.js:1197)
    at proxy (jquery.min.js:10502)
    at Kernel._handle_iopub_message (kernel.js:1224)
    at Kernel._finish_ws_message (kernel.js:1016)
    at kernel.js:1007
connect.js?v=20200327211635:1639 Uncaught TypeError: Cannot read property 'message' of undefined
    at connect.js?v=20200327211635:1639
    at fire (jquery.min.js:3291)
    at Object.fireWith [as rejectWith] (jquery.min.js:3421)
    at process (jquery.min.js:3645)
(anonymous) @ connect.js?v=20200327211635:1639
fire @ jquery.min.js:3291
fireWith @ jquery.min.js:3421
process @ jquery.min.js:3645
setTimeout (async)
(anonymous) @ jquery.min.js:3663
fire @ jquery.min.js:3291
fireWith @ jquery.min.js:3421
fire @ jquery.min.js:3429
fire @ jquery.min.js:3291
fireWith @ jquery.min.js:3421
process @ jquery.min.js:3645
(anonymous) @ jquery.min.js:3655
mightThrow @ jquery.min.js:3557
process @ jquery.min.js:3625
setTimeout (async)
(anonymous) @ jquery.min.js:3663
fire @ jquery.min.js:3291
fireWith @ jquery.min.js:3421
fire @ jquery.min.js:3429
fire @ jquery.min.js:3291
fireWith @ jquery.min.js:3421
deferred.<computed> @ jquery.min.js:3759
handle_output @ rsconnect.js?v=20200327211635:297
Kernel._handle_output_message @ kernel.js:1197
proxy @ jquery.min.js:10502
Kernel._handle_iopub_message @ kernel.js:1224
Kernel._finish_ws_message @ kernel.js:1016
(anonymous) @ kernel.js:1007
Promise.then (async)
Kernel._handle_ws_message @ kernel.js:1007
proxy @ jquery.min.js:10502
```

Internal ticket: https://rstudioide.zendesk.com/agent/tickets/46105

**Comment by scottmmjackson** [comment]
@koverholt can you reproduce this with the latest release?

**Comment by scottmmjackson** [comment]
Let's verify that this wasn't fixed by https://github.com/rstudio/rsconnect-jupyter/pull/234

**Comment by koverholt** [comment]
Isn't 1.3.1.2 the latest release? That's what I used in this issue. Or do you want me to test from master?

**Comment by jskress** [comment]
@koverholt from `master` please.

**Comment by koverholt** [comment]
Not sure if I messed up something when installing from master.

I used the same steps above but ran the following pip install command instead:

```
C:\Users\Administrator\Desktop\python install\Scripts\pip install jupyter git+https://github.com/rstudio/rsconnect-jupyter
```

followed by the usual install/enable commands, which seemed to install correctly, but then I am not seeing the buttons in the modal to create the manifest, and the following shows up in the console:

```
menubar.js:263 actions jupyter-notebook:find-and-replace does not exist, still binding it in case it will be defined later...
MenuBar.bind_events @ menubar.js:263
MenuBar @ menubar.js:59
(anonymous) @ main.js:152
execCb @ require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1693
check @ require.js?v=951f856e81496aaeec2e71a1c2c0d51f:881
(anonymous) @ require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1136
(anonymous) @ require.js?v=951f856e81496aaeec2e71a1c2c0d51f:134
(anonymous) @ require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1186
each @ require.js?v=951f856e81496aaeec2e71a1c2c0d51f:59
emit @ require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1185
check @ require.js?v=951f856e81496aaeec2e71a1c2c0d51f:936
enable @ require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1173
init @ require.js?v=951f856e81496aaeec2e71a1c2c0d51f:786
callGetModule @ require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1200
completeLoad @ require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1587
onScriptLoad @ require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1714
load (async)
req.load @ require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1939
load @ require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1682
load @ require.js?v=951f856e81496aaeec2e71a1c2c0d51f:832
fetch @ require.js?v=951f856e81496aaeec2e71a1c2c0d51f:822
check @ require.js?v=951f856e81496aaeec2e71a1c2c0d51f:854
enable @ require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1173
enable @ require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1554
(anonymous) @ require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1158
(anonymous) @ require.js?v=951f856e81496aaeec2e71a1c2c0d51f:134
each @ require.js?v=951f856e81496aaeec2e71a1c2c0d51f:59
enable @ require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1110
init @ require.js?v=951f856e81496aaeec2e71a1c2c0d51f:786
(anonymous) @ require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1457
setTimeout (async)
req.nextTick @ require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1812
localRequire @ require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1446
requirejs @ require.js?v=951f856e81496aaeec2e71a1c2c0d51f:1794
(anonymous) @ main.js:30
utils.js:60 load_extensions Arguments(2) ["jupyter-js-widgets/extension", "rsconnect_jupyter/index", callee: (...), Symbol(Symbol.iterator): ƒ]
bidi.js:19 Loaded moment locale en
session.js:54 Session: kernel_created (471a9c12-32fc-4aed-99cf-a99cf60b06f2)
kernel.js:462 Starting WebSockets: ws://localhost:8888/api/kernels/4f1ad588-4eee-4e39-8fed-2fdb28d429a9
kernel.js:106 Kernel: kernel_connected (4f1ad588-4eee-4e39-8fed-2fdb28d429a9)
utils.js:37 Loading extension: jupyter-js-widgets/extension
kernel.js:106 Kernel: kernel_ready (4f1ad588-4eee-4e39-8fed-2fdb28d429a9)
utils.js:37 Loading extension: rsconnect_jupyter/index
jquery.min.js:9837 GET http://localhost:8888/nbextensions/rsconnect_jupyter/version.json 404 (Not Found)
send @ jquery.min.js:9837
ajax @ jquery.min.js:9434
ajax @ utils.js:797
(anonymous) @ rsconnect.js?v=20200330215425:526
mightThrow @ jquery.min.js:3557
process @ jquery.min.js:3625
setTimeout (async)
(anonymous) @ jquery.min.js:3663
fire @ jquery.min.js:3291
fireWith @ jquery.min.js:3421
fire @ jquery.min.js:3429
fire @ jquery.min.js:3291
fireWith @ jquery.min.js:3421
done @ jquery.min.js:9533
(anonymous) @ jquery.min.js:9785
load (async)
send @ jquery.min.js:9804
ajax @ jquery.min.js:9434
ajax @ utils.js:797
getVersionInfo @ rsconnect.js?v=20200330215425:522
maybeCreateConfig @ connect.js?v=20200330215425:37
onCreateManifestClicked @ connect.js?v=20200330215425:1794
dispatch @ jquery.min.js:5237
elemData.handle @ jquery.min.js:5044
connect.js?v=20200330215425:1898 Uncaught TypeError: Cannot read property 'rsconnect_jupyter_server_extension' of null
    at jQuery.fn.init.open (connect.js?v=20200330215425:1898)
    at proxy (jquery.min.js:10502)
    at dialog.js:114
open @ connect.js?v=20200330215425:1898
proxy @ jquery.min.js:10502
(anonymous) @ dialog.js:114
setTimeout (async)
(anonymous) @ dialog.js:111
dispatch @ jquery.min.js:5237
elemData.handle @ jquery.min.js:5044
trigger @ jquery.min.js:8471
(anonymous) @ jquery.min.js:8549
each @ jquery.min.js:367
each @ jquery.min.js:202
trigger @ jquery.min.js:8548
(anonymous) @ bootstrap.min.js:6
fn @ jquery.min.js:4985
handle @ bootstrap.min.js:6
dispatch @ jquery.min.js:5237
elemData.handle @ jquery.min.js:5044
trigger @ jquery.min.js:8471
(anonymous) @ jquery.min.js:8549
each @ jquery.min.js:367
each @ jquery.min.js:202
trigger @ jquery.min.js:8548
(anonymous) @ bootstrap.min.js:6
setTimeout (async)
n.fn.emulateTransitionEnd @ bootstrap.min.js:6
(anonymous) @ bootstrap.min.js:6
fn @ jquery.min.js:4985
handle @ bootstrap.min.js:6
dispatch @ jquery.min.js:5237
elemData.handle @ jquery.min.js:5044
trigger @ jquery.min.js:8471
(anonymous) @ jquery.min.js:8549
each @ jquery.min.js:367
each @ jquery.min.js:202
trigger @ jquery.min.js:8548
(anonymous) @ bootstrap.min.js:6
setTimeout (async)
n.fn.emulateTransitionEnd @ bootstrap.min.js:6
s.backdrop @ bootstrap.min.js:6
s.show @ bootstrap.min.js:6
(anonymous) @ bootstrap.min.js:6
each @ jquery.min.js:367
each @ jquery.min.js:202
r @ bootstrap.min.js:6
modal @ dialog.js:147
onCreateManifestClicked @ connect.js?v=20200330215425:1798
dispatch @ jquery.min.js:5237
elemData.handle @ jquery.min.js:5044
```

Seems like a different/unrelated issue. Anything that I should be doing differently to install from master.

**Comment by scottmmjackson** [comment]
There is definitely something wrong with the installation. You repeated the three installation commands, right?

**Comment by koverholt** [comment]
Confirmed that this issue is fixed now!

<img width="1709" alt="Screen Shot 2020-03-31 at 6 06 25 PM" src="https://user-images.githubusercontent.com/303236/78083207-70915b00-737a-11ea-83bd-f808bf2150e1.png">

I am able to generate a manifest fine when calling to a Python binary that has a space in its path. No errors or console errors.

The issue that I posted in my last comment was an issue on my part, I think that my installation of `rsconnect-python` was messed up somehow. Reinstalling `rsconnect-python`, then reinstalling `rsconnect-jupyter` from master worked fine.

Thanks! 👍
